### PR TITLE
chore(middleware): add guest middleware

### PR DESCRIPTION
**Pull request by RomainLanz**

### CODE REVIEW

File: app/Http/Middleware/RedirectIfAuthenticated.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+   *
+   * @param  {Object}   request  [description]
+   * @param  {Object}   response [description]
+   * @param  {Function} next     [description]
+   * @return {void}
+   *
+   * @public
+   */
+  * handle(request, response, next) {
+    if (request.auth.check()) {
```
Comment (thetutlage) [inline app/Http/Middleware/RedirectIfAuthenticated.js]: `auth.check()` is async so it needs to be 

``` javascript
const isLoggedIn = yield request.auth.check()
if (isLoggedIn) {
 ....
}
```

File: app/Http/kernel.js [hunk 2]
```diff
@@ -29,6 +29,8 @@ const globalMiddleware = [
 |
 */
 const namedMiddleware = {
+  'auth': 'Adonis/Middleware/Auth',
+  'guest': '../Middleware/RedirectIfAuthenticated'
```
Comment (thetutlage) [inline app/Http/kernel.js]: Middleware can be added as using the namespace

``` javascript
'guest': 'App/Http/Middleware/RedirectIfAuthenticated'
```

**Comment by RomainLanz** [comment]
Changes done.

**Comment by RomainLanz** [comment]
I have tried to use it I get the exception `handle does not exists on App/Http/Middleware/RedirectIfAuthenticated`.

**Pull request closed without merging**